irohad/ametsuchi/impl/tx_presence_cache_impl.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ametsuchi/impl/tx_presence_cache_impl.hpp"
+#include "common/visitor.hpp"
+#include "interfaces/iroha_internal/transaction_batch.hpp"
+#include "interfaces/transaction.hpp"
+
+namespace iroha {
+  namespace ametsuchi {
+    TxPresenceCacheImpl::TxPresenceCacheImpl(std::shared_ptr<Storage> storage)
+        : storage_(storage) {}
+
+    TxCacheStatusType TxPresenceCacheImpl::check(
+        const shared_model::crypto::Hash &hash) const {
+      auto res = memory_cache_.findItem(hash);
+      if (res != boost::none) {
+        return res.get();
+      }
+
+      TxCacheStatusType cache_status_check;
```

Comment:
Maybe separate this part into a distinct function? This is not about checking the cache, but making a query to storage, so, I think, something like `checkInStorage(Hash)` will be better for clearness